lib/connection.js

```diff
@@ -0,0 +1,18 @@
+class Connection {
```

Comment:
Considering that this is not a connection per se, but just connection config resolver, I would recommend naming it as such - ConnectionConfigResolver

Reply:
https://github.com/tgriesser/knex/pull/3403/files#diff-cbc1ae6887774cf34c8d4bc73dece12fR14